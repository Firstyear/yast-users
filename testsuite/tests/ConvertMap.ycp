/**
 * Test for UsersLDAP::ConvertMap
 * Author:	Jiri Suchomel <jsuchome@suse.cz>
 * $Id$
 */
{
    // testedfiles: Users.pm UsersLDAP.pm

    import "Testsuite";
    import "Users";
    import "Mode";
    import "Directory";
    import "UsersLDAP";

    map READ = $[
	"product": $[
		"features": $[
		    "USE_DESKTOP_SCHEDULER"	: "no",
		    "IO_SCHEDULER"		: "",
		    "ENABLE_AUTOLOGIN"		: "false",
		    "UI_MODE"			: "simple",
		    "EVMS_CONFIG"		: "no",
		    "INCOMPLETE_TRANSLATION_TRESHOLD"	: "99",
		]
	],
	"ldap" : $[
	    "schema": $[
		"oc": $[
		    "may": [
			"gidNumber", "gecos", "objectClass", "uid", "uidNumber",
			"userPassword", "loginShell"
		    ],
		],
	    ],
	],
    ];
    map WRITE = $[];
    map EXEC = $[];

    Testsuite::Dump ("==========================================================");

    Mode::SetTest ("test");

    map<string,any> user = $[
	"dn"		: "uid=test2,ou=people,dc=suse,dc=cz",
	"gidNumber"	: 100,
	"groupname"	: "users",
	"encrypted"	: true,
	"gecos"		: "test",
	"grouplist"	: $[ "audio" : 1 ],
	"loginShell"	: "/bin/csh",
	"modified"	: "edited",
	"objectClass"	: [ "inetOrgPerson", "posixAccount", "shadowAccount", "top" ],
	"org_uid"	: "test2",
	"org_uidnumber"	: 11111,
	"plugins"	: [ "UsersPluginLDAPAll" ],
	"type"		: "ldap",
	"uid"		: "test2",
	"uidNumber"	: 11111,
	"userPassword"	: "{crypt}IVj2W92x/IuFs",
	"what"		: "edit_user",

	"org_user"	: $[
	    "dn"		: "uid=test2,ou=people,dc=suse,dc=cz",
	    "gidNumber"		: 100,
	    "groupname"		: "users",
	    "gecos"		: "test",
	    "loginShell"	: "/bin/bash",
	    "grouplist"		: $[ "audio" : 1 ],
	    "objectClass"	: [ "inetOrgPerson", "posixAccount", "shadowAccount", "top" ],
	    "type"		: "ldap",
	    "uid"		: "test2",
	    "uidNumber"		: 11111,
	    "userPassword"	: "{crypt}IVj2W92x/IuFs",
	]
    ];

    Testsuite::Dump (sformat ("--------- user (loginshell is changed):\n %1", user));

    map<string,any> converted = (map<string,any>) Testsuite::Test(``(UsersLDAP::ConvertMap (user)), [READ,WRITE,EXEC], 0);

    Testsuite::Dump (sformat ("--------- converted :\n %1", converted));

    Testsuite::Dump ("==========================================================");

    user["userPassword"]	= "qqqqq";

    Testsuite::Dump (sformat ("--------- user (new password):\n %1", user));

    converted = (map<string,any>) Testsuite::Test(``(UsersLDAP::ConvertMap (user)), [READ,WRITE,EXEC], 0);

    Testsuite::Dump (sformat ("--------- converted :\n %1", converted));

    Testsuite::Dump ("==========================================================");

    user["objectClass"]	= [ "inetOrgPerson", "posixAccount", "shadowAccount", "top", "sambaSamAccount" ];
    user["userPassword"]		= converted["userPassword"]:"";
    user["org_user", "userPassword"]	= converted["userPassword"]:"";

    Testsuite::Dump (sformat ("--------- user (new object class):\n %1", user));

    converted = (map<string,any>) Testsuite::Test(``(UsersLDAP::ConvertMap (user)), [READ,WRITE,EXEC], 0);

    Testsuite::Dump (sformat ("--------- converted :\n %1", converted));

    Testsuite::Dump ("==========================================================");

    user	= remove (user, "org_user");

    Testsuite::Dump (sformat ("--------- user (removed 'org_user' submap):\n %1", user));

    converted = (map<string,any>) Testsuite::Test(``(UsersLDAP::ConvertMap (user)), [READ,WRITE,EXEC], 0);

    Testsuite::Dump (sformat ("--------- converted :\n %1", converted));

}
