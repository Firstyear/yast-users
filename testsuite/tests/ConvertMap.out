Dump	==========================================================
Dump	--------- user (loginshell is changed):
Dump	 $["dn":"uid=test2,ou=people,dc=suse,dc=cz", "encrypted":true, "gecos":"test", "gidNumber":100, "grouplist":$["audio":1], "groupname":"users", "loginShell":"/bin/csh", "modified":"edited", "objectClass":["inetOrgPerson", "posixAccount", "shadowAccount", "top"], "org_uid":"test2", "org_uidnumber":11111, "org_user":$["dn":"uid=test2,ou=people,dc=suse,dc=cz", "gecos":"test", "gidNumber":100, "grouplist":$["audio":1], "groupname":"users", "loginShell":"/bin/bash", "objectClass":["inetOrgPerson", "posixAccount", "shadowAccount", "top"], "type":"ldap", "uid":"test2", "uidNumber":11111, "userPassword":"{crypt}IVj2W92x/IuFs"], "plugins":["UsersPluginLDAPAll"], "type":"ldap", "uid":"test2", "uidNumber":11111, "userPassword":"{crypt}IVj2W92x/IuFs", "what":"edit_user"]
Return	$["loginShell":"/bin/csh"]
Dump	--------- converted :
Dump	 $["loginShell":"/bin/csh"]
Dump	==========================================================
Dump	--------- user (new password):
Dump	 $["dn":"uid=test2,ou=people,dc=suse,dc=cz", "encrypted":true, "gecos":"test", "gidNumber":100, "grouplist":$["audio":1], "groupname":"users", "loginShell":"/bin/csh", "modified":"edited", "objectClass":["inetOrgPerson", "posixAccount", "shadowAccount", "top"], "org_uid":"test2", "org_uidnumber":11111, "org_user":$["dn":"uid=test2,ou=people,dc=suse,dc=cz", "gecos":"test", "gidNumber":100, "grouplist":$["audio":1], "groupname":"users", "loginShell":"/bin/bash", "objectClass":["inetOrgPerson", "posixAccount", "shadowAccount", "top"], "type":"ldap", "uid":"test2", "uidNumber":11111, "userPassword":"{crypt}IVj2W92x/IuFs"], "plugins":["UsersPluginLDAPAll"], "type":"ldap", "uid":"test2", "uidNumber":11111, "userPassword":"qqqqq", "what":"edit_user"]
Return	$["loginShell":"/bin/csh", "userPassword":"{crypt}qqqqq"]
Dump	--------- converted :
Dump	 $["loginShell":"/bin/csh", "userPassword":"{crypt}qqqqq"]
Dump	==========================================================
Dump	--------- user (new object class):
Dump	 $["dn":"uid=test2,ou=people,dc=suse,dc=cz", "encrypted":true, "gecos":"test", "gidNumber":100, "grouplist":$["audio":1], "groupname":"users", "loginShell":"/bin/csh", "modified":"edited", "objectClass":["inetOrgPerson", "posixAccount", "shadowAccount", "top", "sambaSamAccount"], "org_uid":"test2", "org_uidnumber":11111, "org_user":$["dn":"uid=test2,ou=people,dc=suse,dc=cz", "gecos":"test", "gidNumber":100, "grouplist":$["audio":1], "groupname":"users", "loginShell":"/bin/bash", "objectClass":["inetOrgPerson", "posixAccount", "shadowAccount", "top"], "type":"ldap", "uid":"test2", "uidNumber":11111, "userPassword":"{crypt}qqqqq"], "plugins":["UsersPluginLDAPAll"], "type":"ldap", "uid":"test2", "uidNumber":11111, "userPassword":"{crypt}qqqqq", "what":"edit_user"]
Return	$["loginShell":"/bin/csh", "objectClass":["inetOrgPerson", "posixAccount", "shadowAccount", "top", "sambaSamAccount"]]
Dump	--------- converted :
Dump	 $["loginShell":"/bin/csh", "objectClass":["inetOrgPerson", "posixAccount", "shadowAccount", "top", "sambaSamAccount"]]
Dump	==========================================================
Dump	--------- user (removed 'org_user' submap):
Dump	 $["dn":"uid=test2,ou=people,dc=suse,dc=cz", "encrypted":true, "gecos":"test", "gidNumber":100, "grouplist":$["audio":1], "groupname":"users", "loginShell":"/bin/csh", "modified":"edited", "objectClass":["inetOrgPerson", "posixAccount", "shadowAccount", "top", "sambaSamAccount"], "org_uid":"test2", "org_uidnumber":11111, "plugins":["UsersPluginLDAPAll"], "type":"ldap", "uid":"test2", "uidNumber":11111, "userPassword":"{crypt}qqqqq", "what":"edit_user"]
Return	$["gecos":"test", "gidNumber":"100", "loginShell":"/bin/csh", "objectClass":["inetOrgPerson", "posixAccount", "shadowAccount", "top", "sambaSamAccount"], "uid":"test2", "uidNumber":"11111", "userPassword":"{crypt}qqqqq"]
Dump	--------- converted :
Dump	 $["gecos":"test", "gidNumber":"100", "loginShell":"/bin/csh", "objectClass":["inetOrgPerson", "posixAccount", "shadowAccount", "top", "sambaSamAccount"], "uid":"test2", "uidNumber":"11111", "userPassword":"{crypt}qqqqq"]
